compiler/merge.cc

```diff
@@ -53,6 +53,71 @@ bool MaybeMergeSplitConcat(Graph* graph, Node* node) {
     return true;
 }
 
+bool MaybeMergePadConv(Graph* graph, Node* pad) {
+    if (pad->value() != 0.0 || pad->mode() != "constant") {
+        return false;
+    }
+
+    auto const& pads = pad->pads();
+
+    // range check
+    if (pads.size() < 4) {
+        return false;
+    }
+
+    // pads should apply to feature dims
+    if (pads[0] != 0 || pads[1] != 0 || pads[pads.size() / 2] != 0 || pads[pads.size() / 2 + 1] != 0) {
+        return false;
+    }
+
+    // pads of begin and end must be same
+    for (auto i = 2; i < pads.size() / 2; ++i) {
+        if (pads[i] != pads[pads.size() / 2 + i]) {
+            return false;
+        }
+    }
+
+    // pads must be greater or equal to 0
+    if (std::any_of(pads.begin(), pads.end(), [](auto p) { return p < 0; })) {
```

Comment:
[optional] This is fine as is, but I feel simple `for` loop is shorter and easy to read.

```
for (int64_t p : pads) {
  if (p < 0) {
    return false;
  }
}
```